app/gui/src/controller/ide/desktop.rs

```diff
@@ -48,44 +48,49 @@ pub struct Handle {
 }
 
 impl Handle {
-    /// Create a project controller handle with already loaded project model.
-    pub fn new_with_project(
+    /// Create IDE controller. If `maybe_project_name` is `Some`, a project with provided name will
+    /// be opened. Otherwise controller will be used for project manager operations by Welcome
+    /// Screen.
+    pub async fn new(
         project_manager: Rc<dyn project_manager::API>,
-        initial_project: model::Project,
-    ) -> Self {
+        maybe_project_name: Option<ProjectName>,
```

Comment:
I would just call it `project_name`. The "maybe" part is indicated by the type.

Reply:
Fixed